dist/src/main/java/io/camunda/application/commons/search/SearchEngineSchemaInitializer.java

```diff
@@ -46,8 +50,52 @@ public void afterPropertiesSet() throws IOException {
                   searchEngineConfiguration,
                   clientAdapter.objectMapper())
               .withMetrics(schemaManagerMetrics);
-      schemaManager.startup();
+      if (!addShutdownHook(executor)) {
+        LOGGER.info("skipping schema initialization, JVM is shutting down");
+        return;
+      }
+      executor.submit(schemaManager::startup).get();
+    } catch (final InterruptedException ie) {
+      LOGGER.warn("Schema initialization task was interrupted");
+      LOGGER.debug("Stack trace:", ie);
+      Thread.currentThread().interrupt();
+      return;
+    } catch (final Exception e) {
+      if (isShutdown.get()) {
+        LOGGER.warn("Schema initialization interrupted with shutdown. Message: {}", e.getMessage());
+        LOGGER.debug("Stack trace:", e);
+        Thread.currentThread().interrupt();
+        return;
+      }
+      throw e;
     }
     LOGGER.info("Search engine schema initialization complete.");
   }
+
+  /**
+   * Adds a shutdown hook to the JVM that will be triggered when the application is shutting
+   *
+   * @return true if the shutdown hook was added successfully, false if the JVM is already shutting
+   *     down
+   */
+  private boolean addShutdownHook(final ExecutorService executor) {
+    try {
+      Runtime.getRuntime()
+          .addShutdownHook(
+              new Thread(
+                  () -> {
+                    LOGGER.debug("Shutdown hook triggered");
+                    if (isShutdown.compareAndSet(false, true)) {
+                      executor.shutdownNow();
+                    }
+                  }));
+      return true;
+    } catch (final IllegalStateException e) {
+      // This can happen if the shutdown hook is added after the JVM has started shutting down.
+      // In this case, we just ignore the exception.
+      LOGGER.info("JVM is shutting down, cannot add shutdown hook: {}", e.getMessage());
```

Comment:
:x: I don't think this is interesting for users to know - this is more debug info, isn't it? Or do you think it's useful because it will explain the warning they might later get? It would also be useful to give more details about what the impact of not adding the shutdown hook is if possible.

See https://github.com/camunda/camunda/blob/main/docs/observability/logging.md#info

I would propose

```java
LOGGER.debug("JVM is shutting down, cannot add the schema initializer shutdown hook", e);
```